### ISSUE TITLE
Auto-purchase prioritize player slots

### ISSUE THREAD

**Issue by Lordeath19** [dcs-liberation/dcs_liberation#2842]
**Is your feature request related to a problem? Please describe.**
Most of the time playing on liberation, Me and my buddies are going on different airframes (I go F-16, buddy goes F-18, third buddy goes M-2000 and so on)

In the end we have around 6 airframes that usually are a "must" to continue the campaign.
When auto-purchase is enabled. we find ourselves cutting the AI's purchases to buy our own airframes back.
After a few turns, we usually just disable the auto-purchase as it is a hassle to continue selling AI aircraft and buying our aircrafts.

**Describe the solution you'd like**
Adding a "Player pilots get priority over auto-purchase" option. Basically, make the AI first purchase all destroyed "Player" aircraft. Only after having 1 plane for each player slot can it resume buying what it needs to fulfill the auto-tasks.

**Additional context**
No additional context.

**Comment by DanAlbert** [comment]
Purchase decisions are 100% driven by planning decisions. Is this option not working for you? If so, refile as a bug so we get the data we need to investigate.


![image](https://user-images.githubusercontent.com/315852/236648009-d31b0ec8-a848-476c-be4e-70d03995eb6d.png)

**Comment by Lordeath19** [comment]
I didn't explain my self properly.

The automatic package planning, in `prefer player pliots` behavior from what I found follows this logic (correct me if I'm wrong):

1. Find best squadron (this selects an **aircraft type**)
2. Use a player pilot for chosen **aircraft type** or if none exist use an AI pilot
3. Build the package

The issue in this logic for me is in this scenario:
I play with a carrier, containing 24 F-14s, and an airbase for me and my friends containing 2xSu-27, F-16, F-15.
The AP (automatic planner) tries to create BARCAP flights. follows the logic from before and finds:
1. Find best squadron for CAP (Chooses the F-14 airframe)
2. Fails finding a player pilot (we don't fly the F-14), Instead succeeds finding an AI pilot
3. Builds a BARCAP flight containing 2xF-14 with AI flights and 0 of our custom aircraft (that we do want to fly).

In short, the AP creates BARCAP flights without us because we don't have the best aircraft the task.

The following scenario leads to the issue I described in the title where:
We frag ourselves in a custom BARCAP plan.
Something happens and some of us die (lets say the F-16 dies).
In the next turn, the AP does the same routine, and chooses the F-14 again.
because it completed the BARCAP request and didn't need any F-16 for BARCAP (as it found the F-14), no purchase request is made and the F-16 is never bought back again.

I hope I explained the situation a bit better, I can instead open this as a bug in the AP if you wish

**Comment by DanAlbert** [comment]
Hmm. That sounds very plausible. I don't know if that's actually helpful behavior though, because I expect anyone that uses that really means "make sure all players have a mission every turn".

I think doing better actually recently became a lot easier. I suspect players that are in a squadron with BARCAP enabled want the planner to put them in BARCAP _occasionally_ but not every turn. Now that the primary mission type of the squadron will be preferred, I think that's possible. Alternatively maybe we need to do something weird like plan everything ignoring players, then replace _some_ flight with players.

The options are a bit funky, but I agree that the behavior you describe isn't all that useful.
